src/dataframe/dataframe.jl

```diff
@@ -724,6 +745,8 @@ for T1 in (:AbstractVector, :Not, :Colon),
         end
         for (j, col) in enumerate(idxs)
             df[row_inds, col] = new_df[!, j]
+            # copy column metadata from source to target data frame
```

Comment:
@nalimilan - since this is `setindex!` involving data frame on RHS do you think we should sync table-level metadata (as in other places - to only leave metadata that is present in both tables), or leave LHS table-level metadata unchanged?

Reply:
Hard to tell... We could consider that it's an operation involving two tables, which is literally the case, and therefore sync table-level metadata. Or say that for consistency with other `setindex!` methods it's better to treat this as calling `setindex!` individually on each column, and therefore not sync it. :-/ Maybe the latter is more expected?